From e71cf4e42662daf2b94abf8dceb533a106166118 Mon Sep 17 00:00:00 2001
Message-Id: <e71cf4e42662daf2b94abf8dceb533a106166118.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Mon, 7 Jul 2014 15:16:10 +0100
Subject: [PATCH 140/312] FOR_UPSTREAM [VPG]: drm/i915: Add
 WaAllocateSLML3CacheCtrlOverride:bdw

Allocate SLM WA to fix media resets via L3 cache control register on
BDW.

Issue: VIZ-4086
Change-Id: I35b1521174bca4434a17eee3aa2b05cf4cd9b7d4
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    5 +++++
 drivers/gpu/drm/i915/intel_pm.c |    6 ++++++
 2 files changed, 11 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 9075185..d0abceb 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4734,6 +4734,11 @@ enum punit_power_well {
 #define GEN7_L3SQCREG4				0xb034
 #define  L3SQ_URB_READ_CAM_MATCH_DISABLE	(1<<27)
 
+#define GEN8_L3CNTLREG				0x7034
+#define  GEN8_L3CNTLREG_ALL_L3_CLIENT_POOL	(48<<25)
+#define  GEN8_L3CNTLREG_URB_ALLOCATION		(16<<1)
+#define  GEN8_L3CNTLREG_SLM_MODE_ENABLE		(1<<0)
+
 /* GEN8 chicken */
 #define HDC_CHICKEN0				0x7300
 #define  HDC_FENCE_DESTINATION_TO_SLM_DISABLE	(1<<14)
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 48f6882..f2ce500 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5627,6 +5627,12 @@ static void gen8_init_clock_gating(struct drm_device *dev)
 	I915_WRITE(GEN8_L3CNTLREG1, (I915_READ(GEN8_L3CNTLREG1) &
 			GEN8_TAG_CLK_OFFTIME_MASK) |
 			GEN8_TAG_CLK_OFFTIME);
+
+	/* WaAllocateSLML3CacheCtrlOverride:bdw */
+	I915_WRITE(GEN8_L3CNTLREG,
+			GEN8_L3CNTLREG_ALL_L3_CLIENT_POOL |
+			GEN8_L3CNTLREG_URB_ALLOCATION |
+			GEN8_L3CNTLREG_SLM_MODE_ENABLE);
 }
 
 static void haswell_init_clock_gating(struct drm_device *dev)
-- 
1.7.9.5

